### PR TITLE
[TS] LPS-200548

### PR DESCRIPTION
**Pull request by MirnaGama**
### Context:
[LPS-200548 - NullPointerException in LocalizationImpl when doing checkSystemRoles](https://liferay.atlassian.net/browse/LPS-200548)

### Reproducing Steps:

1. Add the property system.roles=My System Role to portal-ext.properties.
2. Start Liferay. 

✅ Expected: There are no errors in the logs. 
❌ Observed: There's an error in the log:

```
2023-10-31 10:53:48.939 ERROR [main][LocalizationImpl:903] null
java.lang.NullPointerException: null
        at com.ctc.wstx.sw.BaseStreamWriter.writeCharacters(BaseStreamWriter.java:498) ~[wstx.jar:3.2.8]
        at com.liferay.portal.util.LocalizationImpl.getXml(LocalizationImpl.java:889) ~[portal-impl.jar:?]
        at com.liferay.portal.util.LocalizationImpl.getXml(LocalizationImpl.java:831) ~[portal-impl.jar:?]
        at com.liferay.portal.kernel.util.LocalizationUtil.getXml(LocalizationUtil.java:288) ~[portal-kernel.jar:?]
        at com.liferay.portal.service.impl.RoleLocalServiceImpl.checkSystemRoles(RoleLocalServiceImpl.java:333) ~[portal-impl.jar:?]
```

**_Note:_** The same problem will happen with the properties system.organization.roles and system.site.role.

### Implementation details

This exception is thrown when PropsUtil.get() returns null in _checkSystemRoles_ method while fetching the system properties for _system.organization.roles_, _system.site.role_ and _system.roles_. To resolve this, a new method has been added to check the values ​​and will return a blank string if they are null.

### CODE REVIEW

**Comment by MirnaGama** [comment]
ci:test:relevant

**Comment by victoroscuro** [comment]
ci:test:um

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java [hunk 1]
```diff
@@ -2149,6 +2135,20 @@ protected void validate(
 		}
 	}
 
+	private String _getPropertySystemDescription(String name, String property) {
```
Comment (smottal) [inline portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:2138]: I will call it only `_getPropertyDescription`

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java [hunk 2]
```diff
@@ -2149,6 +2135,20 @@ protected void validate(
...
+		String propertySystemDescription = PropsUtil.get(
```
Comment (smottal) [inline portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:2139]: Same here, only `String propertyDescription`

**Review by smottal — CHANGES_REQUESTED** [review]
Hi @MirnaGama, just two very little things on this. Thx

**Comment by MirnaGama** [comment]
Done @smottal !
Please check: https://github.com/liferay-commerce/liferay-portal/pull/4396/commits/7f47725052588d394612c59d1de148effd420d1d

File: portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java (hunk 2 above)
Comment (smottal) [reply portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java:2138]: @MirnaGama could you please change also the name of the variable? Thx

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by MirnaGama** [comment]
Done ! @smottal

**Review by smottal — APPROVED** [review]
LGTM! Thx

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:um

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:um

**Comment by jaydawu** [comment]
ci:test:stable

**Comment by jaydawu** [comment]
There are no related errors to this pull, it can be force-forwarded, thanks!
cc. @victoroscuro

**Pull request closed without merging**